src/functions/containers/FunctionEditPage.tsx

```diff
@@ -1,21 +1,107 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, lazy, Suspense, useContext, useEffect, useState} from 'react'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
 
 // Components
-import {Page} from '@influxdata/clockface'
+import {
+  Page,
+  RemoteDataState,
+  SpinnerContainer,
+  TechnoSpinner,
+} from '@influxdata/clockface'
+import FunctionHeader from 'src/functions/components/FunctionHeader'
+import FunctionForm from 'src/functions/components/FunctionForm'
+import FunctionListProvider, {
+  FunctionListContext,
+} from 'src/functions/context/function.list'
+
+const FluxMonacoEditor = lazy(() =>
+  import('src/shared/components/FluxMonacoEditor')
+)
 
 // Utils
 import {pageTitleSuffixer} from 'src/shared/utils/pageTitles'
+import {getOrg} from 'src/organizations/selectors'
 
 const FunctionEditPage: FC = () => {
+  const history = useHistory()
+  const {id: functionID} = useParams<{id: string}>()
+  const {id: orgID} = useSelector(getOrg)
+
+  const [name, setName] = useState('New Function')
+  const [description, setDescription] = useState('')
+  const [params, setParams] = useState('')
+  const [script, setScript] = useState('')
+
+  const {functionsList} = useContext(FunctionListContext)
+  useEffect(() => {
+    const {name, script, description} = functionsList[functionID]
```

Comment:
this kind of feels like the `this.setState(this.props)` pattern that causes us a bunch of problems (cause we seem to forget edge cases of mutations usually), but we also don't have a draft pattern yet...